genesis_core/user_api/iam/api/controllers.py

```diff
@@ -44,6 +47,47 @@ def enforce(self, rule, do_raise=False, exc=None):
         return iam.enforcer.enforce(rule, do_raise, exc)
 
 
+class ValidateMixin:
+    min_length = 8
+    not_contain: list[str] = [string.whitespace]
+    must_contain: list[str] = None  # [digits, ascii_uppercase, punctuation]
+    regex: str = None
+
+    def validate(self, value):
+        error = None
+        if value is None:
+            error = "Value is required"
+        elif self.min_length and len(value) < self.min_length:
+            error = f"Value must be at least {self.min_length} characters long"
+        elif self.not_contain:
+            for not_contain in self.not_contain:
+                if set(not_contain) & set(value):
+                    error = f"Value must not contain {self.not_contain}"
+                    break
+        elif self.must_contain:
+            for required in self.must_contain:
+                if not set(required) & set(value):
+                    error = f"Value must contain one of {required}"
+                    break
+        elif self.regex and not re.match(self.regex, value):
+            error = f"Value must match regex {self.regex}"
+        if error:
+            exc = ValidationErrorException()
+            exc.message = error
+            raise exc
```

Comment:
why not:
```
if error:
    raise ValidationErrorException(error)
```

Reply:
Maybe just create new exception?

Example: 
```
class ValidationPropertyIncompatibleError(ValidationErrorException):

    message = "%(val)s is not compatible with model %(model)s"

```

So you can push kwarg inside as message parameter (see examples in restalchemy)